tensorflow_io/core/python/experimental/elasticsearch_dataset_ops.py

```diff
@@ -19,72 +19,175 @@
+            try:
+                resource, columns, raw_dtypes = core_ops.io_elasticsearch_readable_init(
+                    healthcheck_url=healthcheck_url,
+                    healthcheck_field="status",
+                    request_url=request_url,
+                )
+                print("Connection successful: {}".format(healthcheck_url))
+                dtypes = []
+                for dtype in raw_dtypes:
+                    if dtype == "DT_INT32":
+                        dtypes.append(tf.int32)
+                    elif dtype == "DT_INT64":
+                        dtypes.append(tf.int64)
+                    elif dtype == "DT_DOUBLE":
+                        dtypes.append(tf.double)
+                    elif dtype == "DT_STRING":
+                        dtypes.append(tf.string)
+                return resource, columns, dtypes, request_url
+            except Exception:
+                print("Skipping host: {}".format(healthcheck_url))
+                continue
+        else:
+            raise ConnectionError(
+                "No healthy node available for this index, check the cluster status and index"
+            )
+
+    def prepare_next_batch(self, resource, columns, dtypes, request_url):
+        """Prepares the next batch of data based on the request url and
+        the counter index.
+
+        Args:
+            resource: the init op resource.
+            columns: list of columns to prepare the structured data.
+            dtypes: tf.dtypes of the columns.
+            request_url: The request url to fetch the data
+        Returns:
+            Structured data which columns as keys and the corresponding tensors as values.
+        """
+
+        url_obj = urlparse(request_url)
+        scroll_request_url = "{}://{}/_search/scroll".format(
+            url_obj.scheme, url_obj.netloc
+        )
+
+        values = core_ops.io_elasticsearch_readable_next(
+            resource=resource,
+            request_url=request_url,
+            scroll_request_url=scroll_request_url,
+            dtypes=dtypes,
+        )
+        data = {}
+        for i, column in enumerate(columns.numpy()):
+            data[column.decode("utf-8")] = values[i]
+        return data
+
+
+class ElasticsearchIODataset(tf.compat.v2.data.Dataset):
     """Represents an elasticsearch based tf.data.Dataset"""
 
-    def __init__(self, hosts=None, protocol="http", internal=True):
+    def __init__(self, nodes, index, doc_type=None, protocol="http", internal=True):
+        """Prepare the ElasticsearchIODataset.
+
+        Args:
+            nodes: A `tf.string` tensor containing the hostnames of nodes
+                in [hostname:port] format. For example: ["localhost:9200"]
+            index: A `tf.string` representing the elasticsearch index to query.
+            doc_type: A `tf.string` representing the type of documents in the index
+                to query.
+            protocol: An optional `tf.string` representing the protocol to
```

Comment:
@kvignesh1420 I am wondering if it makes sense to merge `protocol` with `nodes`, so that the nodes will be:
```
http[s]://localhost:9200
http[s]://ip0:9200,ip1:9200,ip2:9200
```

Reply:
addressed in the commit and updated PR description.

Reply:
This reduces the additional arg and confusion as to what default protocol is being used. Thanks.